### ISSUE TITLE
Exception - Attempt to assign property "company" on bool

### ISSUE THREAD

**Issue by sheetal071** [iomad/iomad#1778]
Iomad version - 4.1.2
PHP version - 8.0
Database	mysql - 8.0.32
While creating a new user we got an exception - Attempt to assign property "company" on bool

![image](https://user-images.githubusercontent.com/108781673/225559405-5974735d-7c74-47d5-8bc6-aefaf671ea7a.png)

**Comment by ni-ght** [comment]
Hello, same here, when clicking: Site administration > Users > Accounts > Add a new user

The "Add a new user" button has the associated URL: moodle_url('/user/editadvanced.php', array('id' => -1));
'-1' I think indicates that a user is not yet associated with the "user_editadvanced_form".

In "/local/iomad/lib/iomad.php", "function profile_definition", $userid == -1 is passed to the function "company::get_company_byuserid($userid)", which then returns a bool.

Perhaps one possible solution is to add a condition check to the "function profile_definition" in "/user/profile/lib.php":

```
// IOMAD - Filter categories which only apply to this company.
if ($userid !== -1) {
    $categories = iomad::iomad_filter_profile_categories($categories, $userid);
}
```

Many thanks.

**Comment by mikeaveli007** [comment]
> Hello, same here, when clicking: Site administration > Users > Accounts > Add a new user
> 
> The "Add a new user" button has the associated URL: moodle_url('/user/editadvanced.php', array('id' => -1)); '-1' I think indicates that a user is not yet associated with the "user_editadvanced_form".
> 
> In "/local/iomad/lib/iomad.php", "function profile_definition", $userid == -1 is passed to the function "company::get_company_byuserid($userid)", which then returns a bool.
> 
> Perhaps one possible solution is to add a condition check to the "function profile_definition" in "/user/profile/lib.php":
> 
> ```
> // IOMAD - Filter categories which only apply to this company.
> if ($userid !== -1) {
>     $categories = iomad::iomad_filter_profile_categories($categories, $userid);
> }
> ```
> 
> Many thanks.

Getting the same error and came to the same solution as a workaround but what is interesting is it works on another instance I have setup with the same exact version.  The error makes sense when you follow the code, what doesn't make sense is why it works on my working instance lol.

**Comment by mikeaveli007** [comment]
I think this has to do with PHP 8.  I checked my env for the working system (PHP 7.4) and prod system (PHP 8.0), noticed there were warnings for that same line on the PHP 7.4 system in the error log and realized I had different PHP versions.

**Comment by egyptrise52** [comment]
Hello I,m a beginner and need your help regarding this error where I can find the file and how to edit it thanks in advance

**Comment by egyptrise52** [comment]
Hello I,m a beginner and need your help regarding this error where I can find the file, and how to edit it thanks in advance

**Comment by egyptrise52** [comment]
> Hello, same here, when clicking: Site administration > Users > Accounts > Add a new user
> 
> The "Add a new user" button has the associated URL: moodle_url('/user/editadvanced.php', array('id' => -1)); '-1' I think indicates that a user is not yet associated with the "user_editadvanced_form".
> 
> In "/local/iomad/lib/iomad.php", "function profile_definition", $userid == -1 is passed to the function "company::get_company_byuserid($userid)", which then returns a bool.
> 
> Perhaps one possible solution is to add a condition check to the "function profile_definition" in "/user/profile/lib.php":
> 
> ```
> // IOMAD - Filter categories which only apply to this company.
> if ($userid !== -1) {
>     $categories = iomad::iomad_filter_profile_categories($categories, $userid);
> }
> ```
> 
> Many thanks.

I just replace that code with the one mentioned above and its works  thanks a lot

**Comment by turf212** [comment]
this was fixed in c37d6dc39c6b
